packages/pds/src/actor-store/blob/reader.ts

```diff
@@ -73,4 +73,62 @@ export class BlobReader {
       ? { applied: true, ref: res.takedownRef }
       : { applied: false }
   }
+
+  async getRecordsForBlob(cid: CID): Promise<string[]> {
+    const res = await this.db.db
+      .selectFrom('record_blob')
+      .where('blobCid', '=', cid.toString())
+      .selectAll()
+      .execute()
+    return res.map((row) => row.recordUri)
+  }
+
+  async blobCount(): Promise<number> {
+    const res = await this.db.db
+      .selectFrom('blob')
+      .select(countAll.as('count'))
+      .executeTakeFirst()
+    return res?.count ?? 0
+  }
+
+  async recordBlobCount(): Promise<number> {
+    const res = await this.db.db
+      .selectFrom(
+        this.db.db
+          .selectFrom('record_blob')
+          .select('blobCid')
+          .distinct()
+          .as('unique_cids'),
+      )
+      .select(countAll.as('count'))
+      .executeTakeFirst()
+    return res?.count ?? 0
```

Comment:
This could be simplified down to `select count(distinct "blobCid") from record_blob`.